File: src/main/java/com/severett/chargerapp/data/SessionStatisticsRepo.java
package com.severett.chargerapp.data;

import java.time.Instant;

public interface SessionStatisticsRepo {

    void addSessionStart(Instant timestamp);

    void addSessionStop(Instant timestamp);

    long getSessionStartCount(Instant beginTimestamp, Instant endTimestamp);

    long getSessionStopCount(Instant beginTimestamp, Instant endTimestamp);

}

File: src/main/kotlin/com/severett/chargerapp/data/SessionStatisticsRepo.kt
package com.severett.chargerapp.data

import java.time.Instant

interface SessionStatisticsRepo {

    fun addSessionStart(timestamp: Instant)

    fun addSessionStop(timestamp: Instant)

    fun getSessionStartCount(fromTimestamp: Instant, toTimestamp: Instant): Long

    fun getSessionStopCount(fromTimestamp: Instant, toTimestamp: Instant): Long

}
